web/src/users/add_url.rs

```diff
@@ -0,0 +1,32 @@
+use axum::{
+    extract::{Form, State},
+    response::{IntoResponse, Redirect},
+};
+use db::{urls::add_url, users::User};
+use serde::Deserialize;
+
+use crate::AppState;
+
+#[derive(Deserialize)]
+pub struct ArticleForm {
+    content: String,
+}
+
+#[axum::debug_handler(state = AppState)]
+pub async fn insert_article_handler(
+    State(state): State<AppState>,
+    user: User,
+    Form(form): Form<ArticleForm>,
+) -> impl IntoResponse {
+    let url = form.content;
+    let user_id = user.user_id;
+
+    println!("inserting");
+    match add_url(&state.db, &url, user_id, &true).await {
+        Ok(_) => Redirect::to("/dashboard"),
```

Comment:
Nice! Eventually I think this might redirect to a page about the specific article we added but we can do that next 💯

Reply:
And actually since this is a redirect to the page you are already on, want to add a flash message here too?

Might need to extend the Flash stuff to have a `success` type instead of the error. Up for giving it a shot?

Reply:
Yea absolutely!